grails-app/assets/javascripts/spApp/directive/selectSpecies.js

```diff
@@ -150,9 +151,48 @@
                             LayoutService.openModal('tool', {processName: 'ToolAddSpeciesService'})
                         };
 
+                        // Old values only has the change of 'changing' item in AngularJs<1.7
+                        // thus cannot simply rollback to old values
+                        // scope.$watchGroup(['spatiallyValid','spatiallySuspect','spatiallyUnknown'], function(newValues, oldValues) {
+                        //     if(newValues.every(function(x) { return !x; })){
+                        //         alert('Select at least one spatial related options!')
+                        //         scope.spatiallyValid = oldValues[0];
+                        //         scope.spatiallySuspect = oldValues[1];
+                        //         scope.spatiallyUnknown = oldValues[2];
+                        //     }
+                        // });
+
+                        scope.$watch('spatiallyValid', function (newVal,oldVal) {
+                            if (!newVal && !scope.validateSpatiallyOptions()) {
+                                alert("Select at least one spatial related options!")
```

Comment:
I think these alert statements should have i18n lookups??? Same for lines 173 and 179 below.